agent/engine/containermetadata/metadata_manager_unix.go

```diff
@@ -0,0 +1,75 @@
+// +build !windows
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+)
+
+const (
+	mountPoint = "/ecs/metadata"
+)
+
+// CreateMetadata creates the metadata file and adds the metadata directory to
+// the container's mounted host volumes
+// binds []string is passed by value to avoid race conditions by multiple
+// calls to CreateMetadata, although this should never actually happen
+func (manager *metadataManager) CreateMetadata(binds []string, task *api.Task, container *api.Container) ([]string, error) {
+	// Do not create metadata file for internal containers
+	// Add error handling for this case? Probably no need since
+	// Internal containers should not be visible to users anyways
+	if container.IsInternal {
+		return binds, nil
+	}
+
+	// Create task and container directories if they do not yet exist
+	mdDirectoryPath, err := getMetadataFilePath(task, container, manager.cfg.DataDir)
+	// Stop metadata creation if path is malformed for any reason
+	if err != nil {
+		return binds, fmt.Errorf("Invalid metadata file path with error: %v", err)
```

Comment:
I do not think we can return nil here because the binds could have values previously present and we do not want to overwrite anything else just because creating the metadata failed.